echopype/convert/set_groups_azfp.py

```diff
@@ -124,6 +124,22 @@ def set_beam(self) -> xr.Dataset:
         freq = np.array(unpacked_data["frequency"]) * 1000  # Frequency in Hz
         ping_time = self.parser_obj.ping_time
 
+        # creates unique channel name
+        if unpacked_data["serial_number"].size == 1:
+            freq_as_str = unpacked_data["frequency"].astype(int).astype(str)
+
+            # TODO: replace str(i+1) with Frequency Number from XML
+            channel_id = [
+                str(unpacked_data["serial_number"]) + "-" + freq_as_str[i] + "-" + str(i + 1)
+                for i in range(len(freq_as_str))
+            ]
+        else:
+            channel_id = None
+            raise NotImplementedError(
+                "Creating a channel name for more than"
+                + " one serial number has not been implemented."
+            )
+
```

Comment:
seems this block could be factored out to be a private method and used by both `set_beam` and `set_vendor` below?

Reply:
Absolutely, I should have done that. @leewujung thank you for pointing that out.

Reply:
I have pushed changes that implement what you have suggested @leewujung.